includes/class-wc-post-data.php

```diff
@@ -490,6 +491,24 @@ public static function force_default_term( $object_id, $terms, $tt_ids, $taxonom
 		}
 	}
 
+	/**
+	 * Ensure statuses are correctly reassigned when restoring orders and products.
+	 *
+	 * @param string $new_status      The new status of the post being restored.
+	 * @param int    $post_id         The ID of the post being restored.
+	 * @param string $previous_status The status of the post at the point where it was trashed.
+	 * @return string
+	 */
+	public function wp_untrash_post_status( $new_status, $post_id, $previous_status ) {
```

Comment:
This function needs to be declared as static as it is being called statically. Otherwise you will get notices in logs.

Reply:
Just pushed this fix, thanks for the review <3